paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkTypeUtils.java

```diff
@@ -173,8 +173,7 @@ public DataType visit(RowType rowType) {
             List<StructField> fields = new ArrayList<>(rowType.getFieldCount());
             for (DataField field : rowType.getFields()) {
                 StructField structField =
-                        DataTypes.createStructField(
-                                field.name(), field.type().accept(this), field.type().isNullable());
+                        DataTypes.createStructField(field.name(), field.type().accept(this), true);
```

Comment:
This change is for https://github.com/apache/incubator-paimon/issues/1048. Should submit another pr for this, just leave it for the pr's tests.